fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -269,6 +271,19 @@ ErrorOr<Optional<TenantMapEntry>> getTenantEntry(ProxyCommitData* commitData,
 	return Optional<TenantMapEntry>();
 }
 
+ErrorOr<Optional<TenantName>> getTenantName(ProxyCommitData* commitData, int64_t tenantId, bool logOnFailure) {
+	auto itr = commitData->tenantIdNameMap.find(tenantId);
+	if (itr == commitData->tenantIdNameMap.end()) {
+		if (logOnFailure) {
+			TraceEvent(SevWarn, "CommitProxyUnknownTenant", commitData->dbgid).detail("TenantId", tenantId);
+		}
+
+		return unknown_tenant();
```

Comment:
The `unknown_tenant` error is a retryable one meant to cause a transaction to retry if the tenant is not present when it was at the start of the transaction. If you want to fail the transaction, you would use `tenant_not_found`.

I'm not certain which is more appropriate here, but for a raw access transaction it seems less likely that the problem would be fixed (or detected) by a retry.